src/main/java/bio/terra/cli/command/app/passthrough/Git.java

```diff
@@ -44,20 +54,36 @@ protected void executeImpl() {
           "Conflicted input argument. Only specify one depending on "
               + "whether you'd like to clone all or some of the git repo resources in this workspace");
     }
-    if (cloneAll) {
+    // handle terra git clone
+    if (cloneAll || (names != null && names.length > 0)) {
       validateCloneCommand();
-      Context.requireWorkspace().getResources().stream()
-          .filter(resource -> Resource.Type.GIT_REPO == resource.getResourceType())
-          .forEach(resource -> cloneGitRepoResource(resource));
-      return;
-    }
-    if (names != null) {
-      validateCloneCommand();
-      for (String name : names) {
-        cloneGitRepoResource(Context.requireWorkspace().getResource(name));
+
+      List<Resource> gitReposToClone = new ArrayList<>();
+      if (cloneAll) {
+        var resources = Context.requireWorkspace().getResources();
+        gitReposToClone.addAll(getGitReposToClone(resources));
```

Comment:
You created a helper method (getGitReposToClone) for only 3 lines. I think it would make this code easier to read, if there were helper methods for more lines.

```
     if (cloneAll) {
       gitReposToClone = getGitReposInWorkspace();
     } else {
       gitReposToClone = getGitReposFromNamesFlag();
     }
```

Also, `getGitReposToClone` isn't really accurate, it should be `getGitRepoResources`.